### ISSUE TITLE
Bug: GridFieldAddExistingAutoCompleter options overflow page when at bottom of tab

### ISSUE THREAD

**Issue by davejtoews** [silverstripe/silverstripe-framework#9377]
## Affected Version

SilverStripe 4.4

## Description

When an autocompleter is the last item on the page, the autocomplete options, which always show beneath the text field, overflow the tab area, leaving all but the first unclickable. Options should appear above the field when field is near bottom of tab, or even near bottom of window.
<img width="1246" alt="Screen Shot 2020-01-17 at 5 01 55 PM" src="https://user-images.githubusercontent.com/9664035/72654257-0278f280-394c-11ea-8152-d8af231c1d36.png">

## Steps to Reproduce

In the attached screenshot the field is placed in the 'buttons-after-left' position of a `GridFieldButtonRow('after')`. When searching for items, the autocomplete results flow past the end of the tab.

**Comment by robbieaverill** [comment]
Thanks for the bug report. To clarify, do you expect the list to be above the input field rather than below it when there isn't enough space in the viewport for them to render?

On a side note, this field has been in need of a React rewrite for some time, and I think it's unlikely that the core team will have a chance to look at this bug before that happens. That said, it's something that can be considered when designing the new field, and we always welcome community contributions as well.

**Comment by davejtoews** [comment]
Yes I was thinking the list should be above the field when there is no space to render.
